### PR TITLE
Refactor iterators (compose SimplePolarIndexer), fix I/O bugs, add 48 tests

### PR DESCRIPTION
**Pull request by sakoho81**
### Iterators refactor + bug fixes
- **Rewrite iterators to compose `SimplePolarIndexer`** — eliminates duplicated meshgrid/r computation. `SimplePolarIndexer` now exposes `self.meshgrid`.
- **6 bugs fixed:**
  - `SectionedFourierRingIterator`: initial angle sector used `d_bin` (spatial) instead of `d_angle` (degrees) — sector width was potentially 10x wrong
  - `AxialExcludeSectionedFourierShellIterator`: `d_extract_angle` parameter silently ignored (never passed to parent)
  - `SectionedFourierShellIterator`: `rotation_stop = 360/d_angle - 1` was float, causing off-by-one on non-divisible angles
  - `HollowSectioned...` + `AxialExclude...`: `np.logical_xor` → `full_section & ~extract_section` (xor is incorrect when extract extends beyond full)
  - `filters.py`: `enumerate(iterator)` unpacking was wrong — `ring` was a tuple, not indices
- **Cleanup:** removed `ring_start`/`shell_start` dead attributes, added shape assertions, lazy SimpleITK import, docstrings
- **24 new tests** covering all 7 iterator classes

### I/O critical bug fixes
- `FourierCorrelationDataWriter`: `for key, value in metadata` → `metadata.items()` (was silently unpacking dict keys)
- `FourierCorrelationDataWriter`: group creation was inside `if exists` block — data only written on overwrite
- `FourierCorrelationDataWriter`: `%d` → `%g` for fractional spacing (e.g. 0.123 μm became 0)
- `FourierCorrelationDataReader`: spacing strings now `float()` converted, not kept as strings
- `pyimq.py`: `read.get_image(path)` → `read.get_image(real_path)` (directory vs file)
- `read.py`/`write.py`: bare asserts → proper exceptions, extension validation, dead code removed
- **9 new tests**: metadata roundtrip, image roundtrip, spacing float preservation, dataset roundtrip, duplicate detection

### Store/operations deeper tests
- **15 new tests**: `chunk_size`, `get_registered_block` edge cases (fully outside, negative start), numpy scalar attr conversion, multi-channel independence, facade methods (`add_psf`, `add_transform`, context manager), `create_rescaled_images` downscale

### Total: 388 tests (295 original + 93 new), ruff+mypy clean

### CODE REVIEW

File: miplib/data/io/fourier_correlation_data_reader.py [hunk 1]
```diff
@@ -43,15 +41,20 @@ def read_images(self, index=None):
             raise ValueError("No images to read")
 
         if index is not None:
-            image_name = "image_%i" % index
-            data_set = self.data["images"][image_name]
-            spacing = data_set.attrs["pixel_size"].split()[0 : len(data_set.shape)]
-            return Image(data_set[:], spacing)
+            image_name = f"image_{index}"
```
Comment (sakoho81) [inline miplib/data/io/fourier_correlation_data_reader.py]: We can reduce repetition here. Also its a bit weird that the return type changes depending on whether you have an index or not. Should it maybe be indices?

File: miplib/data/io/fourier_correlation_data_writer.py [hunk 2]
```diff
@@ -32,102 +27,84 @@ def __init__(self, output_dir, filename, append=False):
...
-        """
-        Write a metadata dictionary to the HDF5 files header as a general description of
-        the dataset.
-        :param metadata: a dictionary with all the necessary data to be written in the file
-        """
-        assert isinstance(metadata, dict)
-
-        for key, value in metadata:
+        """Write a metadata dictionary to the HDF5 file header attributes."""
+        if not isinstance(metadata, dict):
```
Comment (sakoho81) [inline miplib/data/io/fourier_correlation_data_writer.py]: lets use a static check here. We should add proper type annotations, an remove these useless isinstance checks.

File: miplib/data/io/fourier_correlation_data_writer.py [hunk 3]
```diff
@@ -32,102 +27,84 @@ def __init__(self, output_dir, filename, append=False):
...
             self.data.attrs[key] = value
 
     def write_images(self, images):
-        """
-        Write images to the data structure
-        :param images: a tuple of Image objects, or a single image
-        """
+        """Write Image objects to the data structure."""
         if not isinstance(images, tuple):
-            images = tuple(images)
```
Comment (sakoho81) [inline miplib/data/io/fourier_correlation_data_writer.py:54]: why is it important that we have a tuple here? Also type annotations are missing.

File: miplib/data/io/fourier_correlation_data_writer.py [hunk 4]
```diff
@@ -32,102 +27,84 @@ def __init__(self, output_dir, filename, append=False):
...
-                )
-                resolution_group.attrs["resolution"] = data_set.resolution["resolution"]
-                resolution_group.attrs["resolution-point"] = (
-                    "%d %d (yx)" % data_set.resolution["resolution-point"]
-                )
-                resolution_group.attrs["criterion"] = data_set.resolution["criterion"]
-                resolution_group.create_dataset(
-                    "resolution-threshold-coefficients",
-                    data=data_set.resolution["resolution-threshold-coefficients"],
+                raise ValueError(
```
Comment (sakoho81) [inline miplib/data/io/fourier_correlation_data_writer.py]: do we want to crash here? Maybe we can skip or overwrite based on toggle?

File: miplib/data/io/read.py [hunk 5]
```diff
@@ -27,6 +27,10 @@ def get_image(filename, series=0, channel=0, return_type="image", bioformats=Tru
     if filename.endswith(".mha"):
         data = __itk_image(filename, return_type == "itk")
     else:
+        if not filename.endswith((".tif", ".tiff")):
```
Comment (sakoho81) [inline miplib/data/io/read.py]: This is not correct. THe bioformats can read pretty much any kind of an image. Maybe lets let it worry about the validation?

File: miplib/data/io/read.py [hunk 6]
```diff
@@ -83,16 +87,25 @@ def __tiff(filename, memmap=False, return_itk=False):
 
     # Figure out z-spacing, which in ImageJ is hidden in the "image_description"
     # header (why, one might ask).
+    if "image_description" not in tags:
```
Comment (sakoho81) [inline miplib/data/io/read.py]: Maybe we can just log an error,  and use 1 for default spacing? Of coruse then the physical dimensions are off. What do you think?

File: miplib/data/io/read.py [hunk 7]
```diff
@@ -158,9 +171,8 @@ def __bioformats(filename, series=0, channel=0, return_itk=False):
     :param return_itk:
     :return:
     """
-    assert pims.bioformats.available(), (
-        "Please install jpype in order to use the bioformats reader."
-    )
+    if not pims.bioformats.available():
```
Comment (sakoho81) [inline miplib/data/io/read.py:64]: can we ensure that the setup is complete in uv? Do we need external dependencies?

File: miplib/data/io/write.py [hunk 8]
```diff
@@ -6,34 +6,28 @@
...
 
     :param path:    A full path to the image.
-    :param image:   An image as :type image: numpy.ndarray or sitk.image.
-
-    :return:
+    :param image:   An Image (ndarray subclass with spacing).
     """
-
-    assert isinstance(image, Image)
+    if not isinstance(image, Image):
```
Comment (sakoho81) [inline miplib/data/io/write.py]: type annotations missing throughout your changes

File: miplib/data/io/write.py [hunk 9]
```diff
@@ -6,34 +6,28 @@
...
-    A writer for ITK supported image formats.
-
-    :param path:    A full path to the image.
-    :param image:   An image as :type image: numpy.ndarray.
-    :param spacing: Pixel size ZXY, as a :type spacing: list.
-    """
-    assert isinstance(image, Image)
-
+    """Write an Image in ITK format (.mha/.mhd)."""
+    if not isinstance(image, Image):
```
Comment (sakoho81) [inline miplib/data/io/write.py]: also here, I would just go with static checks

File: miplib/data/iterators/fourier_ring_iterators.py [hunk 10]
```diff
@@ -105,14 +86,9 @@ def angle(self, value):
         self.angle_sector = self.get_angle_sector(angle, angle + self.d_angle)
 
     def get_angle_sector(self, phi_min, phi_max):
```
Comment (sakoho81) [inline miplib/data/iterators/fourier_ring_iterators.py]: type annotations missing

File: miplib/data/iterators/fourier_shell_iterators.py [hunk 11]
```diff
@@ -291,71 +210,54 @@ def get_angle_sector(self, phi_min, phi_max):
...
-        """
-        :param shape: Shape of the data
-        :param d_bin: The radius increment size (pixels)
-        :param d_angle: The angle increment size (degrees)
-        """
+        if len(shape) != 3:
+            raise ValueError(f"shape must be 3D, got shape {shape}")
 
-        assert len(shape) == 3, "This iterator assumes a 3D shape"
+        import miplib.processing.itk as itkutils
```
Comment (sakoho81) [inline miplib/data/iterators/fourier_shell_iterators.py]: why are these not on top?
